src/utils/progress_utils.py

```
@@ -0,0 +1,207 @@
+"""
+Progress reporting utilities for MCP tools.
+"""
+import asyncio
+from typing import Optional
```

Ruff (0.12.2)

`typing.Optional` imported but unused

Remove unused import: `typing.Optional`

(F401)